### PR TITLE
Apiserver authentication

### PR DESCRIPTION
**Pull request by filipmatusak**

### CODE REVIEW

File: authproxy/app/helpers_test.go [hunk 1]
```diff
@@ -44,3 +44,26 @@ func cleanData(t *testing.T) func() {
...
+	bMap := make(map[string]int)
+
+	for _, a := range as {
+		aMap[a]++
+	}
+	for _, b := range bs {
+		bMap[b]++
+	}
+
+	for a, count := range aMap {
```
Comment (samo98) [inline authproxy/app/helpers_test.go:63]: can't we replace this for-cycle with `reflect.DeepEqual(aMap, bMap)`?

File: authproxy/app/user.go [hunk 2]
```diff
@@ -78,3 +78,21 @@ func (a *App) updateUser(res http.ResponseWriter, req *http.Request) {
 
 	handlers.RespondWithJSON(res, http.StatusOK, user)
 }
+
+func (a *App) getEmails(res http.ResponseWriter, req *http.Request) {
+	vatNumbers := req.URL.Query()["vatNumber"]
+	taxIds := req.URL.Query()["taxId"]
+
+	if len(vatNumbers)+len(taxIds) == 0 {
+		handlers.RespondWithError(res, http.StatusBadRequest, "There should be at least 1 email requested")
```
Comment (samo98) [inline authproxy/app/user.go:87]: this message should be probably `There should be at least 1 vatNumber or taxId requested`

File: authproxy/app/user_test.go [hunk 3]
```diff
@@ -103,3 +104,58 @@ func TestPatchUser(t *testing.T) {
...
+
+	user1.Email = stringPointer("e1")
+	user1.VatNumber = stringPointer("v1")
+	user1.TaxId = stringPointer("t1")
+	user2.Email = stringPointer("e2")
+	user2.VatNumber = stringPointer("v2")
+	user2.TaxId = stringPointer("t2")
+
+	a.manager.UpdateUser(user1)
+	a.manager.UpdateUser(user2)
```
Comment (samo98) [inline authproxy/app/user_test.go:122]: let's create 3rd user without e-mail set and add test where we ensure, empty string/null won't be returned

File: authproxy/db/user.go [hunk 4]
```diff
@@ -57,3 +58,26 @@ func (connector *Connector) CreateUser(user *User) (*User, error) {
...
+
+	if len(vatNumbers) != 0 {
+		query = query.WhereOr("vat_number IN (?)", pg.In(vatNumbers))
+	}
+
+	if len(taxIds) != 0 {
+		query = query.WhereOr("tax_id IN (?)", pg.In(taxIds))
+	}
+
+	err := query.Select(&emails)
```
Comment (samo98) [inline authproxy/db/user.go:76]: probably empty emails filtering is missing

**Review by samo98 — APPROVED** [review]
